### ISSUE TITLE
Gradients do not show correctly in Editor

### ISSUE THREAD

**Issue by james7132** [vexe/VFW#4]
I'm trying to add gradients to some of my objects that currently can be only serialized/edited via VFW. However, it seems that the drawer for gradients does not work.

I know that the Gradient drawer is not available via normal EditorGUI calls and only through EditorGUI.PropertyField on a SerializedProperty of Gradient type.

So my question is: are we to just wait until UT adds available API calls or is there a way to work around this?

EDIT: The serialization data seems to be getting it's data just fine, but the Editor just draws it as a generic object without any visible members.

**Comment by vexe** [comment]
The problem is that the DoGradientField (internal static in EditorGUI) only deals with SerializedProperty which Vfw doesn't play with. I'll see what I can do.

**Comment by vexe** [comment]
Wait, actually I think the SerializedProperty is optional, it does seem necessary however for when you do a right click and copy/paste the gradient value. But that's easily implemented

**Comment by james7132** [comment]
Check #6, I think I have at least a temporary solution working. There are a few issues that may need to be ironed out like the size of the field, but for the most part it seems to be working.

I'm using some reflection for it that could probably be rewritten using Fasterflect, but I don't know how to use it, so I leave that up to you or whoever wants to work on it.

**Comment by vexe** [comment]
Duh! You beat me to it! :) - I actually missed GradientField, I was working on integrating DoGradientField but the former is much easier! 

Btw there's no more Fasterflect, I wrote my own Fast.Reflection which does similar but a lot simpler. All you do is just call DelegateForCall on your MethodInfo which gives back a MethodCaller< TTarget, TReturn > - if you don't know the target type or if there's none (static method) just pass 'object'

**Comment by vexe** [comment]
I removed the empty catch since I didn't get any exceptions, can't see a reason to suppress it.

The reason the field was big was because you were using kMultilineOption which is used for Rects, Bounds and anything that takes two lines to be drawn.

**Comment by james7132** [comment]
OK, makes sense. I'm just trying to figure out how to use the entire framework. It's just sort of hard to follow everything without full documentation. I'm really thankful for the framework though. It's great work. The fact that it can serialize complex object graphs and interfaces within behaviors is saving me a bunch of time and headache already. 

Just a side question: custom drawers should work anywhere in the Editor folders, right? or do they need to be in a specific folder?

**Comment by vexe** [comment]
They should work anywhere in Editor folders yeah. Take a look at the static constructor of MemberDrawersHandler.
